tests/utils/test_eloop.c

```diff
@@ -0,0 +1,169 @@
+#define _GNU_SOURCE
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <string.h>
+#include <inttypes.h>
+#include <unistd.h>
+#include <setjmp.h>
+#include <stdint.h>
+#include <cmocka.h>
```

Comment:
I'm removed a bunch of includes that I think are unnecessary, and added `<sys/un.h>` for Unix domain socket support.

FreeBSD usually requires everything is defined explicitly (and in the correct order), which makes things fun :) (and POSIX compliant)

```suggestion
#include <stdarg.h>
#include <stddef.h>
#include <setjmp.h>
#include <stdint.h>
#include <cmocka.h>

#include <sys/socket.h>
#include <sys/un.h>
#include <stdio.h>
#include <string.h>
```